src/coreComponents/linearAlgebra/interfaces/hypre/HypreMGR.hpp

```diff
@@ -213,6 +213,7 @@ class MGRStrategyBase
 #if GEOS_USE_HYPRE_DEVICE == GEOS_USE_HYPRE_CUDA || GEOS_USE_HYPRE_DEVICE == GEOS_USE_HYPRE_HIP
     GEOS_LAI_CHECK_ERROR( HYPRE_BoomerAMGSetAggInterpType( solver.ptr, hypre::getAMGAggressiveInterpolationType( LinearSolverParameters::AMG::AggInterpType::modifiedExtendedE ) ) );
     GEOS_LAI_CHECK_ERROR( HYPRE_BoomerAMGSetCoarsenType( solver.ptr, hypre::getAMGCoarseningType( LinearSolverParameters::AMG::CoarseningType::PMIS ) ) );
+    GEOS_LAI_CHECK_ERROR( HYPRE_BoomerAMGSetAggNumLevels( solver.ptr, 0 ) );
```

Comment:
can you remove this and create a PR only with this change? @victorapm we should test this. Unless you know why this would impact GPU performance so much.

Reply:
Agreed on testing this. It seems aggressive coarsening did not behave well for some of the problems tested in this PR, while it works well for other problems I tested. For better robustness, we can leave it turned off by default as proposed in this PR, but ultimately we should add the option of controlling this via input file